go-controller/pkg/ovn/gateway_init.go

```diff
@@ -59,9 +59,23 @@ func (oc *DefaultNetworkController) cleanupStalePodSNATs(nodeName string, nodeIP
 	podIPsOnNode := sets.NewString() // collects all podIPs on node
 	for _, pod := range pods.Items {
 		pod := pod
+		if !util.PodScheduled(&pod) || util.PodCompleted(&pod) {
+			if util.PodCompleted(&pod) {
+				collidingPod, err := oc.findPodWithIPAddresses([]net.IP{utilnet.ParseIPSloppy(pod.Status.PodIP)}) //even if a pod is completed we should still delete the nat if the ip is not in use anymore
+				if err != nil {
+					return fmt.Errorf("lookup for pods with same ip as %s failed: %w", pod.Name, err)
+				}
+				if collidingPod != nil { //if the ip is in use we should not remove the nat
+					continue
+				}
+			}
+			if !util.PodScheduled(&pod) {
```

Comment:
hmm this feels strange, keep the check for `PodScheduled` different from `PodCompleted` ? do this block before L62 and that will ensure we don't have to check this twice